ops/testing.py

```diff
@@ -260,7 +260,20 @@ def update_relation_data(self, relation_id, app_or_unit, key_values):
             if rel_data is not None:
                 # If we have read and cached this data, make sure we invalidate it
                 rel_data._invalidate()
-        # TODO: we only need to trigger relation_changed if it is a remote app or unit
+
+        is_peer = self._charm.meta.relations[relation_name].role == 'peers'
+        if self._model.unit == entity:
+            # Updates to our own unit do not trigger relation-changed events.
+            return
```

Comment:
IOW, this would become an error (use relation.data[charm.model.unit][foo] = bar)

I do think we still need a way to initialize the fields, but maybe that wouldn't be *this* function.

Reply:
Agreed, ditto (https://github.com/canonical/operator/pull/212#discussion_r402756835)